src/lib/gadgets/gadgets.ts

```diff
@@ -59,4 +58,37 @@ const Gadgets = {
   xor(a: Field, b: Field, length: number) {
     return xor(a, b, length);
   },
+  /**
+   * Bitwise AND gadget on {@link Field} elements. Equivalent to the [bitwise AND `&` operator in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_AND).
+   * An AND gate works by comparing two bits and returning `1` if both bits are `1`, and `0` otherwise.
+   *
+   * It can be checked by a double generic gate the that verifies the following relationship between the values below (in the process it also invokes the {@link Gadgets.xor} gadget which will create additional constraints depending on `length`).
+   *
+   * The generic gate verifies:\
+   * `a + b = sum` and the conjunction equation `2 * and = sum - xor`\
+   * Where:\
+   * `a + b = sum`\
+   * `a x b = xor`\
+   * `a ^ b = and`
+   *
+   * You can find more details about the implementation in the [Mina book](https://o1-labs.github.io/proof-systems/specs/kimchi.html?highlight=gates#and)
+   *
+   * The `length` parameter lets you define how many bits should be compared. `length` is rounded to the nearest multiple of 16, `paddedLength = ceil(length / 16) * 16`, and both input values are constrained to fit into `paddedLength` bits. The output is guaranteed to have at most `paddedLength` bits as well.
```

Comment:
This is not super accurate. If you write some tests with lengths which are not a multiple of 16, you should still see that the AND behaves as expected. Meaning, if you want to do AND of 7 bits, you can do it. By the way you write "the output is guaranteed to have at most `paddedLength` bits as well", it seems to indicate that the AND operation will be padded to 16 bits, instead of 7.

What I am saying is that this padding information might be too low level for the docs. It's true that the Xor gadget pads to nearest (largest or equal) multiple of 16 bits, filling with zeros in more significant positions. But it seems too unrelated to the And operation to be mentioned here, in my opinion.

Reply:
I think there's a case for only allowing multiples of 16 as length input, to avoid the implied ambiguity. But that would be for another PR